receiver/aerospikereceiver/scraper_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aerospikereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver"
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	as "github.com/aerospike/aerospike-client-go/v5"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver/internal/model"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/aerospikereceiver/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/zap"
+)
+
+func TestNewAerospikeReceiver_BadEndpoint(t *testing.T) {
+	testCases := []struct {
+		name     string
+		endpoint string
+		errMsg   string
+	}{
+		{
+			name:     "no port",
+			endpoint: "localhost",
+			errMsg:   "missing port in address",
+		},
+		{
+			name:     "no address",
+			endpoint: "",
+			errMsg:   "missing port in address",
+		},
+	}
+
+	cs, err := consumer.NewMetrics(func(ctx context.Context, ld pmetric.Metrics) error { return nil })
+	require.NoError(t, err)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cfg := &Config{Endpoint: tc.endpoint}
+			receiver, err := newAerospikeReceiver(component.ReceiverCreateSettings{}, cfg, cs)
+			require.ErrorContains(t, err, tc.errMsg)
+			require.Nil(t, receiver)
+		})
+	}
+}
+
+func TestScrapeNode(t *testing.T) {
```

Comment:
We should have a test for the `scrape` function of the receiver as a whole, too.

Reply:
I think this will have to be an integration test since new connections are created to the discovered nodes. I think I should be able to update the integration tests with this, though

Reply:
You could do that, or make the newASClient function mockable (e.g. function pointer on the scraper that defaults to the actual implementation). Either way, I think it's fine as long is the case is covered by a test.

Reply:
Oh good call, I'll add it to the receiver struct if that's an Ok pattern